apps/ejabberd/src/ejabberd_admin.erl

```diff
@@ -197,7 +197,7 @@ join_cluster(Node) ->
 
 -spec maybe_leave_cluster() -> {ok, string()} | {error, term()} | {not_in_cluster, string()}.
 maybe_leave_cluster() ->
-  NodeList = nodes(),
+  NodeList = mnesia:system_info(running_db_nodes),
```

Comment:
I'm not sure this can happen or not, but what if due to some inconsistency (network issue) we're not in `running_db_nodes`, but are in `db_nodes`? Again, maybe it's impossible, but some food for thought.